lib/experimental/connection-sources/index.php

```diff
@@ -12,7 +12,7 @@
 		// if it doesn't, `get_post_meta()` will just return an empty string.
 		return get_post_meta( $block_instance->context['postId'], $meta_field, true );
 	},
-	'pattern_attributes' => function ( $block_instance, $meta_field ) {
-		return _wp_array_get( $block_instance->context, array( 'dynamicContent', $meta_field ), false );
+	'pattern_attributes' => function ( $block_instance ) {
+		return _wp_array_get( $block_instance->context, array( 'dynamicContent', $block_instance->attributes['metadata']['id'] ), false );
```

Comment:
Nitpicky, but the code could be broken onto a few lines to make it more readable. Maybe store the instance `id` in a variable, and add an inline comment to explain what's happening.

Reply:
The `'name' => 'meta'` thing is confusing me too, but doesn't need to be addressed here, just an observation.

Reply:
The function signature of this "source" file is still TBD I believe. Might be worth to do some more tidying up after the API is stable enough.